src/test/scala/mesosphere/marathon/api/EndpointsHelperTest.scala

```diff
@@ -138,40 +172,94 @@ class EndpointsHelperTest extends UnitTest {
-          requirePorts = true,
-          container = singleServicePort.container.map { c =>
-            c.copyWith(c.portMappings.map(_.copy(servicePort = 88, hostPort = None)))
-          }
-        ),
-        servicePort = 88,
-        hostPort = 0 // weird edge case for tasks on container networks without hostPort defined
-      )
+      s"handle single instance with 1 (static) service port $servicePort and no host port by outputting the container ip and container port" in {
+        val report = parseOutput(EndpointsHelper.appsToEndpointString(ListTasks(instances(appWithoutHostPorts, Seq(1)), Seq(appWithoutHostPorts)), allContainerNetworks))
+        val expected = List(("foo", servicePort.toString, Seq(s"1.1.1.11:8080")))
+        report should equal(expected)
+      }
+
+      s"handle multiple instances, different agents and no host port mapping by outputting the container ip and container port" in {
+        val report = parseOutput(EndpointsHelper.appsToEndpointString(ListTasks(instances(appWithoutHostPorts, Seq(1, 1, 1)), Seq(appWithoutHostPorts)), allContainerNetworks))
+        val expected = List(("foo", "80", List("1.1.1.11:8080", "1.1.1.21:8080", "1.1.1.31:8080")))
+        report should equal(expected)
+      }
+
+      "exclude container-networked endpoints when not included in the network list" in {
+        val report = parseOutput(EndpointsHelper.appsToEndpointString(ListTasks(instances(appWithoutHostPorts, Seq(1)), Seq(appWithoutHostPorts)), noContainerNetworks))
+        val expected = List(("foo", servicePort.toString, Seq()))
+        report should equal(expected)
+      }
+
+      "include only container-networked endpoints that are in the network list" in {
+        val dualServicePorts = fakeApp(
+          container = Option(container.copy(
+            portMappings = Seq(
+              Container.PortMapping(servicePort = 80, hostPort = None, containerPort = 8080, networkNames = Seq(containerNetwork.name)),
+              Container.PortMapping(servicePort = 81, hostPort = None, containerPort = 8081, networkNames = Seq(containerNetwork2.name)),
+            )
+          )),
+          networks = Seq(containerNetwork, containerNetwork2)
+        ).copy(
+          requirePorts = true
+        )
+
+        val report = parseOutput(EndpointsHelper.appsToEndpointString(ListTasks(instances(dualServicePorts, Seq(1)), Seq(dualServicePorts)), Set(containerNetwork.name)))
+        inside(report) {
+          case (app1, port1, mappings1) :: (app2, port2, mappings2) :: Nil =>
+            app1 shouldBe "foo"
+            port1 shouldBe "80"
+            mappings1 shouldBe List("1.1.1.11:8080") // we include containerNetwork port mappings
+
+            app2 shouldBe "foo"
+            port2 shouldBe "81"
+            mappings2 shouldBe List() // but not containerNetwork2 port mappings
+        }
+      }
+
+      s"1.5: handle single instance and no host port by outputting the hostname and 0" in {
```

Comment:
How is the `master` version different?

Reply:
It didn't output a nonsensical address (agent hostname from which the endpoint is fundamentally unreachable anyways, with the port of 0)